### PR TITLE
[iOS] Avoid crash setting custom icon in SearchHandler

### LINKED ISSUE #16297: SearchHandler crashes on ItemSource change in iOS
### Description

When I create a SearchHandler to use in iOS, it crashes when I update the ItemsSource when the query changes

image

### Steps to Reproduce

1. Create a File -> New Maui Project
2. Insert a SearchHandler like in the sample project
3. Load Sample Data
4. Start typing. In the example sample, it should reach out to the json file to load data after 3 characters. It will fail. I've seen this happen with real API calls as well. API calls and JSON loading works in Android.
5. Witness a NullRefException (log shows stacktrace)

### Link to public reproduction project repository

https://github.com/edgiardina/MauiBug_SearchHandler_iOSCrash.App

### Version with bug

7.0.86

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

Tested on iOS 16.4 Simulator(s)

### Did you find any workaround?

No. Removing the template doesn't work either b(trying to use DisplayMemberName)

### Relevant log output

```shell
Object reference not set to an instance of an object.
   at Microsoft.Maui.Controls.ViewExtensions.GetParentsPath(Element self)+MoveNext()
   at Microsoft.Maui.Controls.ViewExtensions.FindMauiContext(Element element, Boolean fallbackToAppMauiContext)
   at Microsoft.Maui.Controls.Platform.Compatibility.UIContainerCell..ctor(String cellId, View view, Shell shell, Object context)
   at Microsoft.Maui.Controls.Platform.Compatibility.UIContainerCell..ctor(String cellId, View view)
   at Microsoft.Maui.Controls.Platform.Compatibility.ShellSearchResultsRenderer.GetCell(UITableView tableView, NSIndexPath indexPath)
   at UIKit.UIApplication.UIApplicationMain(Int32 argc, String[] argv, IntPtr principalClassName, IntPtr delegateClassName)
   at UIKit.UIApplication.Main(String[] args, Type principalClass, Type delegateClass)
   at MauiBug_SearchHandler_iOSCrash.Program.Main(String[] args) in C:\Users\ed_gi\source\repos\MauiBug_SearchHandler_iOSCrash\MauiBug_SearchHandler_iOSCrash\Platforms\iOS\Program.cs:line 13
```

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Avoid crash setting custom icon in SearchHandler on iOS.

### Issues Fixed

Fixes #14442
Fixes #16297

### CODE REVIEW

File: src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellPageRendererTracker.cs [hunk 1]
```diff
@@ -782,7 +782,7 @@ void SearchButtonClicked(object sender, EventArgs e)
 
 		void SetSearchBarIcon(UISearchBar searchBar, ImageSource source, UISearchBarIcon icon)
 		{
-			source.LoadImage(source.FindMauiContext(), image =>
+			source.LoadImage(source.FindMauiContext(true), image =>
```
Comment (PureWeen) [inline src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellPageRendererTracker.cs:785]:
```suggestion
			source.LoadImage(source.FindMauiContext() ?? MauiContext, image =>
```

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellPageRendererTracker.cs (hunk 1 above)
Comment (jsuarezruiz) [reply src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellPageRendererTracker.cs:785]: Mnn, the `FindMauiContext `extension method uses the `MauiContext `of the App if it is null, but couldn't the `MauiContext `property in this context be null?

**Review by rmarinho — CHANGES_REQUESTED** [review]
Can you explain how does this fix the issue? 

Can we also add a test?

**Comment by jsuarezruiz** [comment]
> Can you explain how does this fix the issue?
> 
> Can we also add a test?

Sure, the `ClearIcon` ImageSource from the SearchHandler has not Parent. For that reason, trying to get the context is failing. I am just using the bool parameter from the `FindMauiContext` method to get the Application Context if fail.
Another approach is review the SearchHandler and set the correct parents where is necessary. @PureWeen Thoughts?

**Comment by jsuarezruiz** [comment]
This has been pending for a while. From recent conversations with Shane I think this is not the correct fix and we should revisit it. I'm going to close this PR to look for another fix in another PR.

**Pull request closed without merging**